partiql-test-data/fail/parser/primitives/call.ion

```diff
@@ -0,0 +1,174 @@
+substring::[
+    sql92::[
+        parse::{
+            name: "SUBSTRING sql92 syntax missing left paren",
+            statement: "SELECT SUBSTRING FROM 'asdf' FOR 1) FROM foo",
+            assert: {
+                result: ParseError
+            },
+        },
+        parse::{
+            name: "SUBSTRING sql92 syntax missing FROM or comma",
+            statement: "SELECT SUBSTRING('str' 1) FROM foo",
+            assert: {
+                result: ParseError
+            },
+        },
+        parse::{
+            name: "SUBSTRING sql92 syntax without length and missing right paren",
+            statement: "SELECT SUBSTRING('str' FROM 1 FROM foo",
+            assert: {
+                result: ParseError
+            },
+        },
+        parse::{
+            name: "SUBSTRING sql92 syntax with length missing right paren",
+            statement: "SELECT SUBSTRING('str' FROM 1 FOR 1 FROM foo",
+            assert: {
+                result: ParseError
+            },
+        }
+    ],
+    partiql::[
+        parse::{
+            name: "SUBSTRING without length missing right paren",
+            statement: "SELECT SUBSTRING('str', 1 FROM foo",
+            assert: {
+                result: ParseError
+            },
+        },
+        parse::{
+            name: "SUBSTRING missing right paren",
+            statement: "SELECT SUBSTRING('str', 1, 1 FROM foo",
+            assert: {
+                result: ParseError
+            },
+        }
+    ]
+]
+
+trim::[
+    parse::{
+        name: "TRIM no left paren",
+        statement: "TRIM ' ')",
+        assert: {
+            result: ParseError
+        },
+    },
+    parse::{
+        name: "TRIM too many arguments",
+        statement: "TRIM(BOTH ' ' FROM 'test' 2)",
+        assert: {
+            result: ParseError
+        },
+    },
+    parse::{
+        name: "TRIM with BOTH missing FROM",
+        statement: "TRIM(BOTH 'test')",
+        assert: {
+            result: ParseError
+        },
+    },
+    parse::{
+        name: "TRIM and remove without FROM",
+        statement: "trim(both '' 'test')",
+        assert: {
+            result: ParseError
+        },
+    },
+    parse::{
+        name: "TRIM without string",
+        statement: "TRIM(FROM)",
+        assert: {
+            result: ParseError
+        },
+    },
+    parse::{
+        name: "TRIM no args",
+        statement: "TRIM()",
+        assert: {
+            result: ParseError
+        },
+    },
+    parse::{
+        name: "TRIM with TRAILING missing FROM",
+        statement: "TRIM(trailing '')",
```

Comment:
While this is technically not spec-conformant[1], it does not seem ambiguous. And PostGres 9.4-14 seems to support this syntax[2].

Which is not to say that we should support it, but I'm also not sure of the utility of the negative test here.

[1]: https://jakewheat.github.io/sql-overview/sql-92-grammar.html#trim-operands
[2]: https://www.postgresql.org/docs/14/functions-string.html